Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp

```diff
@@ -918,11 +908,16 @@ static ErrorOr<u16> read_effective_chunk_size(JPEGStream& stream)
     return stored_size - 2;
 }
 
-static ErrorOr<void> ensure_quantization_tables_are_present(JPEGLoadingContext& context)
+static ErrorOr<void> ensure_tables_are_present(JPEGLoadingContext& context)
 {
-    for (auto const& component : context.current_scan->components) {
+    auto const& current_scan = context.current_scan.value();
+    for (auto const& component : current_scan.components) {
         if (!context.registered_quantization_tables[component.component.quantization_table_id])
             return Error::from_string_literal("Unknown quantization table id");
+        if (current_scan.spectral_selection_start == 0 && !context.registered_dc_tables[component.dc_destination_id])
+            return Error::from_string_literal("Unable to find corresponding DC table");
+        if (current_scan.spectral_selection_end > 0 && !context.registered_ac_tables[component.ac_destination_id])
```

Comment:
Note to self: The first if checks `…_start` and the second checks `…_end`, so this is good as-is. (I almost left a comment complaining here, since I missed this at first.)